install.sh

```diff
@@ -1,37 +1,68 @@
 #!/usr/bin/env bash
 set -o errexit -o nounset -o pipefail
 
+has() {
+  [[ -z $1 ]] && return 1
+  command -v $1 >/dev/null 2>&1
+}
+
 echo "Starting Fluence installation..."
 
 # Identify OS and architecture
 OS="$(uname -s | tr '[:upper:]' '[:lower:]')"
 ARCH="$(uname -m)"
 
-# Install in default fluence home dir
+# Install in fluence home dir
 FLUENCE_USER_DIR="${FLUENCE_USER_DIR-$HOME/.fluence}"
 
 # Temporary directory creation and cleanup setup
 TEMP="$(mktemp -d)"
 trap "rm -rf '$TEMP'" EXIT INT TERM
 
+# Check that curl is installed
+if ! has "curl"; then
+  echo "'curl' is required for the installation. Please install it and rerun the script"
+  exit 1
+fi
+
+# Check that fluence is not installed
+if has "fluence"; then
+  cat <<ERR
+fluence cli seems to be already installed.
+If it was installed with with npm please unistall it first with 'npm uninstall -g @fluencelabs/cli' then rerun installation script.
+If it was installed with installation script you should use 'fluence update' command to update to latest version.
+ERR
+  exit 1
+fi
+
 # Verify if the cli directory is empty
 if $(ls -1 ${FLUENCE_USER_DIR}/cli &>/dev/null); then
```

Comment:
This statement is TRUE when `${FLUENCE_USER_DIR}/cli` exists and empty.  Generally this statement will be false only when `ls -1` returns error. Maybe would be better to check for files in `${FLUENCE_USER_DIR}/cli` in some direct way?

Reply:
![image](https://github.com/fluencelabs/cli/assets/22859527/9507fe3d-2878-4b47-9722-13733e95a5a8)

Reply:
Nice catch! But I will change wording about directory being empty or not. If this directory exits we can assume that it was created by installation script only.

I can also do something like this
```
if [[ -d "${FLUENCE_USER_DIR}/cli" ]] && [[ -n "$(ls -A ${FLUENCE_USER_DIR}/cli)" ]]; then
    echo "Directory exists and is not empty"
else
    echo "Directory does not exist or is empty"
fi
```

But I think adding more code in that situation is not worth it.